enterprise/management/commands/validate_default_enrollment_intentions.py

```diff
@@ -0,0 +1,146 @@
+            '--delay-minutes',
+            dest='delay_minutes',
+            required=False,
+            type=int,
+            default=30,
+            help="How long after a customer's catalog has been updated are we allowed to evaluate the customer."
+        )
+
+    @property
+    def latest_change_allowed(self):
+        return timezone.now() - timedelta(minutes=self.delay_minutes)
+
+    def handle_default_enrollment_intention(self, customer, intention):
+        """
+        Check that the default enrollment intention's content_key is contained in any of the customer's catalogs.
+
+        Returns:
+            bool: True if the default enrollment intention is valid.
+        """
+        content_metadata = get_and_cache_customer_content_metadata(
+            customer.uuid,
+            intention.content_key,
+        )
+        contained_in_customer_catalogs = bool(content_metadata)
+        if contained_in_customer_catalogs:
+            logger.info(
+                f"handle_default_enrollment_intention(): Default enrollment intention {intention} "
+                "is compatible with the customer's catalogs."
+            )
+        else:
+            logger.error(
+                f"handle_default_enrollment_intention(): Default enrollment intention {intention} "
+                "is NOT compatible with the customer's catalogs."
+            )
+        return contained_in_customer_catalogs
+
+    def handle_customer(self, customer):
+        """
+        Try to evaluate an EnterpriseCustomer for any invalid DefaultEnterpriseEnrollmentIntention records.
+
+        Returns:
+            dict: A structured result object that indicates whether the
+            customer was skipped, and which intentions are invalid.
+        """
+        result = {
+            'skipped': None,
+            'invalid_intentions': None,
+        }
+        if customer.catalogs_modified_latest > self.latest_change_allowed:
+            result['skipped'] = True
+            logger.info(
+                f"handle_customer(): SKIPPING Evaluating default enrollment intentions for customer {customer}."
+            )
+            return result
+        result['skipped'] = False
+        logger.info(f"handle_customer(): Evaluating default enrollment intentions for customer {customer}.")
+        results = {
+            intention: self.handle_default_enrollment_intention(customer, intention)
+            for intention in customer.default_enrollment_intentions.all()
+        }
+        result['invalid_intentions'] = [intention for intention, valid in results.items() if not valid]
+        return result
+
+    def handle(self, *args, **options):
+        self.delay_minutes = options.get("delay_minutes")
+
+        customers_with_intentions = DefaultEnterpriseEnrollmentIntention.objects.values_list(
+            "enterprise_customer__uuid",
+        )
+        customers_with_intentions = [values[0] for values in customers_with_intentions]
```

Comment:
FYI you can pass in `flat=True` to values list if you're only selecting a single column, and that will give you back singles values instead of 1-tuples.